niceml/filechecksumprocessors/zippedcsvtoparqprocessor.py

```diff
@@ -0,0 +1,184 @@
+            output_fs.makedirs(output_path, exist_ok=True)
+            output_files = list_dir(
+                path=output_path,
+                recursive=self.recursive,
+                file_system=output_fs,
+            )
+            output_files = [
+                join_fs_path(output_fs, output_path, output_file)
+                for output_file in output_files
+                if splitext(output_file)[1] == ".parq"
+            ]
+
+        return input_files, output_files
+
+    def generate_batches(
+        self,
+        input_file_list: List[str],
+        changed_files_dict: Dict[str, Dict[str, bool]],
+        output_file_list: Optional[List[str]] = None,
+        force: bool = False,
+    ) -> List[Dict[str, List[str]]]:
+        """
+        The generate_batches function is responsible for generating a list of batches,
+        where each batch is a dictionary with `inputs` as a key, followed by a list of file paths
+
+        Args:
+            input_file_list: List[str]: A list of input file names
+            changed_files_dict: Dict[str: Dict[str:bool]]: Dictionary with the information
+            which files have changed
+            output_file_list: List[str]: A optional list of output file names
+            force: bool: Force the generation of batches even if no files have changed
+
+        Returns:
+            A list of batches, each batch is a dictionary with one key `inputs`
+            and the value is a list of file paths
+        """
+        if not force:
+            input_file_list = [
+                file_name
+                for file_name, changed in changed_files_dict["inputs"].items()
+                if changed
+            ]
+        batches = []
+        for batch_pos in range(0, len(input_file_list), self.batch_size):
+            batches.append(
+                {"inputs": input_file_list[batch_pos : batch_pos + self.batch_size]}
+            )
+        return batches
+
+    def process(self, batch: Dict[str, List[str]]) -> Dict[str, Dict[str, str]]:
+        """
+        The process function takes a batch of files and converts them from CSV to Parquet.
+
+        Args:
+            self: Access the class attributes
+            batch: Dict[str: Pass in the batch of files to be processed
+            List[str]]: Pass in the list of files that are to be processed
+
+        Returns:
+            A dictionary of checksums for each file in `self.output_location`
+            (key = `outputs`) and `self.input_location` (key = `inputs`)
+        """
+        checksums = defaultdict(dict)
+
+        with open_location(self.input_location) as (input_file_system, input_root):
+            for zip_file in tqdm(
+                batch["inputs"], desc="Extract zip files of current batch"
+            ):
+                with input_file_system.open(zip_file) as opened_zip_file:
+                    checksums["inputs"][zip_file] = input_file_system.checksum(zip_file)
```

Comment:
Change `output_file_system.checksum` into `md5_from_file`

Reply:
Done in [cc88d45](https://github.com/codecentric-oss/niceml/pull/44/commits/cc88d45d9db160ca1d28c704ec762f3c6c8024fc)